lisa/analysis/rta.py

```diff
@@ -160,15 +162,28 @@ def get_log_file(self, task):
         """
         return self.perf_data[task]['logfile']
 
-    def get_df(self, task):
+    @memoized
+    def get_df(self, task, start_time=None):
         """
         Return the pandas dataframe with the performance data for the
-        specified task
+        specified task. A start time can be specified as a reference for the
+        first event, for example to align events to a given (see :class:`TraceView`).
 
         :param task: Name of the task that we want the performance dataframe of.
         :type task: str
+
+        :param start_time: The first event time in seconds
+        :type start_time: float
         """
-        return self.perf_data[task]['df']
+        task_df = self.perf_data[task]['df']
+        if not start_time:
+            return task_df
+
+        # Let's keep a copy so that we can still access the original one
+        task_df = task_df.reset_index()
```

Comment:
Is that giving a copy ? The doc seems to not say anything about that:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.reset_index.html

Reply:
No, that comment is not more correct since I've refactored the code... will update it to align it with the intent ;)